File: templates/ros2/__subpath__/src/ros2_cpp_pkg/src/greeter.cpp
#include "ros2_cpp_pkg/greeter.hpp"

#include <string>

namespace ros2_cpp_pkg
{
namespace
{
std::string trim(const std::string & s)
{
  const auto begin = s.find_first_not_of(" \t\n\r");
  if (begin == std::string::npos) {
    return "";
  }
  const auto end = s.find_last_not_of(" \t\n\r");
  return s.substr(begin, end - begin + 1);
}
}  // namespace

std::string greeting(const std::string & name, int count)
{
  std::string who = trim(name);
  if (who.empty()) {
    who = "world";
  }
  return "hello " + who + " #" + std::to_string(count);
}
}  // namespace ros2_cpp_pkg

File: templates/ros2/__subpath__/src/ros2_py_pkg/ros2_py_pkg/greeter.py
"""纯逻辑示例：与 ROS 解耦，故可在任意环境（含 macOS，无需 rclpy）单测。

约定：把可独立验证的业务逻辑放进这类纯函数 / 类，ROS 节点只做收发翻译——
便于 TDD，便于在无 ROS 的机器上快测（见 test/pure/）。
"""

from __future__ import annotations


def greeting(name: str, count: int) -> str:
    """生成第 count 条问候语（纯函数）。"""
    who = name.strip() or "world"
    return f"hello {who} #{count}"
